src/main/java/frc/robot/superstructure/SuperstructureSubsystem.java

```diff
@@ -0,0 +1,100 @@
+import frc.robot.intake.IntakeSubsystem;
+import frc.robot.intake_rollers.IntakeRollersMode;
+import frc.robot.intake_rollers.IntakeRollersSubsystem;
+import frc.robot.queuer.QueuerMode;
+import frc.robot.queuer.QueuerSubsystem;
+import frc.robot.shooter.ShooterSubsystem;
+
+public class SuperstructureSubsystem extends SubsystemBase {
+  private IntakeSubsystem intakeWrist;
+  private IntakeRollersSubsystem intakeRollers;
+  private QueuerSubsystem queuer;
+  private ShooterSubsystem shooter;
+  private RobotIntakeMode intakeMode = RobotIntakeMode.STOPPED;
+  private RobotShooterMode shooterMode = RobotShooterMode.STOPPED;
+  private double savedRPM = 0;
+  private boolean isIntakeHoming = false;
+
+  /** Creates a new SuperstructureSubsystem. */
+  public SuperstructureSubsystem(
+      IntakeSubsystem intake,
+      IntakeRollersSubsystem intakeRollers,
+      QueuerSubsystem queuer,
+      ShooterSubsystem shooter) {
+    this.intakeWrist = intake;
+    this.intakeRollers = intakeRollers;
+    this.queuer = queuer;
+    this.shooter = shooter;
+  }
+
+  @Override
+  public void periodic() {
+
+    // shooter logic
+    if (shooterMode == RobotShooterMode.MANUAL_SPINUP) {
+      this.savedRPM = 2500;
+    } else if (shooterMode == RobotShooterMode.STOPPED) {
+      this.savedRPM = 0;
+    }
+    this.shooter.setRPM(this.savedRPM);
+
+    // intake roller logic
+    if (intakeMode == RobotIntakeMode.INTAKING) {
+      this.intakeRollers.setMode(IntakeRollersMode.INTAKING);
+    } else if (intakeMode == RobotIntakeMode.OUTTAKING) {
+      this.intakeRollers.setMode(IntakeRollersMode.OUTTAKING);
+    } else if (intakeMode == RobotIntakeMode.STOPPED) {
+      this.intakeRollers.setMode(IntakeRollersMode.STOPPED);
+    }
+
+    // queuer logic
+    if (shooterMode == RobotShooterMode.SHOOTING) {
+      this.queuer.setMode(QueuerMode.SHOOT);
+    } else if (intakeMode == RobotIntakeMode.INTAKING) {
+      this.queuer.setMode(QueuerMode.QUEUEING);
+    } else if (intakeMode == RobotIntakeMode.OUTTAKING) {
+      this.queuer.setMode(QueuerMode.EJECT);
+    } else {
+      this.queuer.setMode(QueuerMode.STOPPED);
+    }
+  }
+
+  public void setIntakeMode(RobotIntakeMode intakeMode) {
+    if(!this.isIntakeHoming) {
+      this.intakeMode = intakeMode;
+      if (intakeMode == RobotIntakeMode.INTAKING) {
+        this.intakeWrist.setPosition(IntakeSetting.INTAKING);
+      } else if (intakeMode == RobotIntakeMode.OUTTAKING) {
+        this.intakeWrist.setPosition(IntakeSetting.OUTTAKING);
+      }
+    }
+  }
+
+  public void setIntakePosition(IntakeSetting intakeSetting) {
+    if (!this.isIntakeHoming) {
+      if (this.intakeMode == RobotIntakeMode.STOPPED) {
+        this.intakeWrist.setPosition(intakeSetting);
+      }
+    }
+  }
+
+  public void setShooterMode(RobotShooterMode shooterMode) {
+    this.shooterMode = shooterMode;
+  }
+
+  public void isIntakeHoming(boolean isHoming) {
+    this.isIntakeHoming = isHoming;
+  }
```

Comment:
i would rename this method to make it clearer that this has side effects. perhaps `setIntakeHoming()`) typically methods starting with `is` indicates returning a boolean